src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2790,16 +2819,14 @@ private ImmutableArray<Diagnostic> GetDiagnosticsForMethodBodiesInTree(SyntaxTre
                 hasDeclarationErrors: false,
                 emitMethodBodies: false,
                 diagnostics: bindingDiagnostics,
-                filterOpt: s => IsDefinedOrImplementedInSourceTree(s, tree, span),
+                filterOpt: filterTree is object ? (Predicate<Symbol>?)(s => IsDefinedOrImplementedInSourceTree(s, filterTree!, filterSpan)) : (Predicate<Symbol>?)null,
```

Comment:
>! [](start = 124, length = 1)

Is this null suppression actually necessary? A quick test with a dummy nullable object suggests it shouldn't be.

Reply:
> Is this null suppression actually necessary? A quick test with a dummy nullable object suggests it shouldn't be.

I didn't add it on my own.

---
In reply to: [597065816](https://github.com/dotnet/roslyn/pull/51855#discussion_r597065816) [](ancestors = 597065816)